src/session/src/context.rs

```diff
@@ -43,6 +43,10 @@ const CURSOR_COUNT_WARNING_LIMIT: usize = 10;
 #[builder(build_fn(skip))]
 pub struct QueryContext {
     current_catalog: String,
+    /// mapping of RegionId to SequenceNumber, for snapshot read, meaning that the read should only
+    /// container data that was committed before(and include) the given sequence number
+    /// this field will only be filled if extensions contains a pair of "snapshot_read" and "true"
+    snapshot_seqs: Arc<RwLock<Option<HashMap<u64, u64>>>>,
```

Comment:
Do we need to use `Option`? We can use an empty map if there are no seqs.

Reply:
> Do we need to use `Option`? We can use an empty map if there are no seqs.

previous using Option is to differ this three case apart:
- not need snapshot: None
- need snapshot but haven't get all snapshot ids: Some(empty)
- need snapshot and snapshots' id are available: Some(regions to ids)

I'm not sure now if first two case be distinguishable is useful anymore ,perhaps not?